src/sentry/search/snuba/backend.py

```diff
@@ -49,6 +51,57 @@ def unassigned_filter(unassigned, projects):
     return query
 
 
+def linked_filter(linked, projects):
+    """
+    Builds a filter for whether or not a Group has an issue linked via either
+    a PlatformExternalIssue or an ExternalIssue.
+    """
+    # Note that there is no filter being applied here with `projects`. We rely on
+    # the planner to avoid a seq scan on the PlatformExternalIssue table via
+    # the more obvious semi-join from Group to PlatformExternalIssue. We notably
+    # do not attempt to filter on `projects` as the the path from a
+    # PlatformExternalIssue to the Project.id goes through the Group table, and
+    # thus would require a seq scan on the PlatformExternalIssue table or an index
+    # scan over the Group rows for the given projects, which would negate the
+    # benefits of any indexes used for other filters. This would be more reasonable if
+    # we had a redundant `project_id` (like GroupLink.project_id) on this table.
+    group_ids_with_platform_external_issue = PlatformExternalIssue.objects.values_list(
```

Comment:
@wedamija : thanks! Happy to poke around adding a migration.

In my mind the correct order of steps is:
1. Add `project_id = BoundedBigIntegerField(null=True, db_index=True)` to `PlatformExternalIssue`, and run in production. I'm assuming like GroupLink, this won't be an FK.
2. Update backend to start setting `project_id` column for new `PlatformExternalIssue`s.
3. Run a data migration that backfills `project_id` which then make project_id non-nullable. Since it only has 10K rows, this can be done in a single transaction.
4. Land this PR now that `project_id` is correctly maintained.

Unless there's a sophisticated system for migrations and the best we have is all migrations are run before deploy, it looks like I can break this down into at least two PRs:
- One PR with 1&2 (or split into two PRs)
- After 1&2 are deployed, a PR that does 3&4 can land / deploys (or split into two PRs).

Does that sound right?

Reply:
1+2: https://github.com/getsentry/sentry/pull/21858